vm/storage.go

```diff
@@ -96,16 +113,35 @@ func (vm *VM) shouldComapct(expiryHeight uint64) bool {
 // compaction as storing blocks randomly on-disk (when using [block.ID]).
 func (vm *VM) UpdateLastAccepted(blk *chain.StatelessBlock) error {
 	batch := vm.vmDB.NewBatch()
-	if err := batch.Put(lastAccepted, binary.BigEndian.AppendUint64(nil, blk.Height())); err != nil {
+	bigEndianHeight := binary.BigEndian.AppendUint64(nil, blk.Height())
+	if err := batch.Put(lastAccepted, bigEndianHeight); err != nil {
+		return err
+	}
+	if err := batch.Put(PrefixBlockKey(blk.Height()), blk.Bytes()); err != nil {
 		return err
 	}
-	if err := batch.Put(PrefixBlockHeightKey(blk.Height()), blk.Bytes()); err != nil {
+	if err := batch.Put(PrefixBlockIDHeightKey(blk.ID()), bigEndianHeight); err != nil {
```

Comment:
We need to store this on-disk now so that we can service historical `Get` requests. We make this tradeoff so that we can avoid db compaction (from storing blocks all over disk).